### DISCUSSION TITLE
Can we opt in to using inert atrribute in addition to aria-hidden in FloatingFocusManager?

### DISCUSSION THREAD

**Discussion by mihkeleidast** [floating-ui/floating-ui#3098] [React] [answered]
Currently, FloatingFocusManager adds `aria-hidden` to all elements that should be avoided when a modal dialog (or whatever requires it) is open.

This results in the following markup, in the case of a button:
```
<button type="submit" aria-hidden="true">Submit</button>
```

This markup is effectively invalid, as it adds `aria-hidden` to an otherwise focusable element. I know that there are focus guards rendered, which should prevent user from navigating outside the floating element, but it does not prevent the element from being focused programmatically - calling `button.focus()` on such element still moves the focus to it. This is certainly an edge case, but it can still happen if you don't have control over absolutely every piece of code that runs on your page.

Adding inert to that element would prevent that, focus is not moved if `.focus()` is called on that element:

```
<button type="submit" aria-hidden="true" inert="inert">Submit</button>
```

It's possible to opt in to `inert` currently by disabling `guards`, but that has other functional consequences I would not like.

------------

What is the background on this logic - why prefer aria-hidden over inert at all? Is it related to browser support? If so, always adding inert should work as progressive enhancement.

**Comment by atomiks** [comment, marked as answer]
This should address that: https://github.com/floating-ui/floating-ui/pull/3131

Well, `inert` was really new at the time when the implementation was first made, so it wasn't considered. This also allows for pointer modality without needing to use a backdrop, since nothing is interactive when it's `modal`.
